src/version_set.cc

```diff
@@ -266,6 +274,7 @@ void VersionSet::AddColumnFamilies(
 Status VersionSet::DropColumnFamilies(
     const std::vector<uint32_t>& column_families,
     SequenceNumber obsolete_sequence) {
+  ReadLock l(&mutex_);
```

Comment:
should hold write lock (if we are to use rwlock). This function modifies `obsolete_columns_`.